README.md

```diff
@@ -49,19 +49,46 @@ It will also mix in the following defaults, which should be customized.
 ### LDAP Verifier class
 
 The `Verifier` class has a `verify` function that is the passport verify callback. In this module it gets called after LDAP authentication succeeds. By default it does nothing but you can overwrite it to make furthers validation
-checks. See [examples/app.js](examples/app.js#L56). 
+checks. See [examples/app.js](examples/app.js#L56).
+
+
+### Asynchronous LDAP Strategy configuration
+
+Per request configuration of the LDAP strategy is supported by taking advantage of
+the passport-ldapauth [asynchronous configuration retrieval](https://github.com/vesse/passport-ldapauth#asynchronous-configuration-retrieval)
+feature.
+
+In place of the options a function can be passed when initializing the feathers-authentication-ldap
+module. This function will be called with the authentication request object and it should
+return a promise that resolves the altered options. These altered options will be merged with
+the default and static settings.
+
+I.E. The credentials passed in an authentication request can be used as the bind credentials
+in an Active Directory setup that does not include common bind credentials...
+
+```javascript
+app.configure(authentication(config))
+.configure(jwt())
+.configure(local())
+.configure(ldap( function (req) {
```

Comment:
Hey @ekryski thanks for taking a look.

I'll take a look at the tests and what is needed for the asynchronous configuration.

Keep in mind that the Passport LdapStrategy expects to receive a function in place of static settings when you need dynamic settings per request. At [this line](https://github.com/bnielsen1965/feathers-authentication-ldap/blob/master/src/index.js#L68) the ldapStrategySettings should be an object or function. The req object used in the async function comes from the request inside Passport when the LdapStrategy calls for the asynchronous settings.

I couldn't find any other way other than passing a function into ldap() at initialization. Everything else I tried would only provide a one time settings for the LdapStrategy.